pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -367,10 +367,10 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
-	// failover mons running on host path to use persistent volumes if VolumeClaimTemplate is set and vice versa
+	// failover any mons present in the mon fail over list
 	for _, mon := range c.ClusterInfo.Monitors {
-		if c.HasMonPathChanged(mon.Name) {
-			logger.Infof("fail over mon %q due to change in mon path", mon.Name)
+		if c.monFailoverList.Has(mon.Name) {
+			logger.Infof("fail over mon %q from the mon fail over list %+v", mon.Name, c.monFailoverList)
 			c.failMon(len(c.ClusterInfo.Monitors), desiredMonCount, mon.Name)
```

Comment:
No need to remove the item from the set after failover because we reconcile the operator after failing over every mon.